libs/util/http.client.js

```diff
@@ -103,41 +107,34 @@ function mergeConfig(config) {
         }
 
         if (config.retry) {
-            interval = parseInt(config.retry && config.retry.interval, 10);
+            interval = parseInt(config.retry.interval, 10);
             if (!isNaN(interval) && interval > 0) {
                 cfg.retry.interval = interval;
             }
-            maxRetries = parseInt(config.retry && config.retry.max_retries, 10);
+            maxRetries = parseInt(config.retry.max_retries, 10);
             if (!isNaN(maxRetries) && maxRetries >= 0) {
                 cfg.retry.max_retries = maxRetries;
             }
+            if (config.retry.statusCodes) {
+                cfg.retry.statusCodes = config.retry.statusCodes
+            }
         }
 
         if (config.withCredentials) {
             cfg.withCredentials = config.withCredentials;
         }
-
-        // tmp stores transient state data, such as retry count
```

Comment:
Why was this removed?

Reply:
**Context**: for every retry, we double the interval for the next request. And, if the amount of retries reaches the maximum limit, `fetchr` just give up and returns an error.

**Previous implementation:** `cfg.tmp` was only being used to store the number of attempts and the interval for the next request. Whenever we would call `shouldRetry`, we would mutate this object by increasing the number of attempts and by doubling the interval so the next retry could be properly set.

 What I did was to replace this mutable object with an explicit required `attempt` param in `doXhr` function. `doXhr` is the function that is responsible for attempting the request, calling `shouldRetry` and scheduling a new request attempt if necessary (by calling itself). 

Since `doXhr` has `attempt` as param now, it can just pass `attempt + 1` to the inner `doXhr` call. Also, we can get the new interval value by just doing `Math.pow(2, attempt) * interval`. So no need for a mutable object that is shared between calls.